index.js

```diff
@@ -90,18 +93,7 @@ module.exports = (inputs, patterns, options) => {
 };
 
 module.exports.isMatch = (inputs, patterns, options) => {
-	inputs = sanitizeArray(inputs, 'inputs');
-	patterns = sanitizeArray(patterns, 'patterns');
+	const matching = module.exports(inputs, patterns, options);
 
-	if (patterns.length === 0) {
-		return false;
-	}
-
-	return inputs.some(input => {
-		return patterns.every(pattern => {
-			const regexp = makeRegexp(pattern, options);
-			const matches = regexp.test(input);
-			return regexp.negated ? !matches : matches;
-		});
-	});
+	return matching.length > 0;
```

Comment:
The problem with this new logic here is that it's inefficient. The old way, it short-circuited when it found a match. Now it will do *everything* and then check for a match.

Reply:
Good point. I modified the code. Now isMatch seems to be bit quicker compared to the original implementation, probably because not constructing Regxp repeatedly.